service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -148,78 +157,58 @@ private Map<String, Map<String, String>> prepareFileInputs(
 
     Map<String, Map<String, String>> fileInputsMap = new HashMap<>();
     for (String fileInputName : fileInputNames) {
+      String fileInputValue = userProvidedInputs.get(fileInputName).toString();
+      String sasUrl =
+          retrieveWriteOnlySasUrl(
+              jobId, workspaceId, storageResourceId, fileInputValue, accessToken);
+
       fileInputsMap.put(
           fileInputName,
-          constructSasUrlAndAzcopyCommand(
-              jobId,
-              workspaceId,
-              storageResourceId,
-              userProvidedInputs.get(fileInputName).toString(),
-              accessToken));
+          Map.of(
+              "sasUrl",
+              sasUrl,
+              "azcopyCommand",
+              "azcopy copy %s %s".formatted(fileInputValue, sasUrl)));
     }
 
     return fileInputsMap;
   }
 
-  private Map<String, String> constructSasUrlAndAzcopyCommand(
+  private String retrieveWriteOnlySasUrl(
```

Comment:
should this method live in WorkspaceManagerService instead?

Reply:
yea i would probably move it there or WSM utils next to the WorkspaceManagerService similar to the AppUtils file for cbas